### PR TITLE
chore: clean up theme manager

### PR DESCRIPTION
**Pull request by benlife5**
- uses MutationObserver api to update styles
- moves all console logs to devLogger (should work with Brian's query params changes)
- removes puckRendered since we did not end up using it

### CODE REVIEW

File: src/components/Editor.tsx [hunk 1]
```diff
@@ -510,7 +510,7 @@ export const Editor = ({
   );
 
   useReceiveMessage("getThemeData", TARGET_ORIGINS, (send, payload) => {
-    console.log(`payload in getThemeData is ${JSON.stringify(payload)}`);
+    devLogger.log("getThemeData PAYLOAD: " + JSON.stringify(payload));
```
Comment (mkilpatrick) [inline src/components/Editor.tsx:513]: In the other useReceiveMessages we have ie `devLogger.logData("SAVE_STATE", payload);`, can you do the same? And then remove the devlogger below that logs it again after it's json escaped. We don't really need it twice. Same in the others.

File: src/internal/components/InternalEditor.tsx [hunk 2]
```diff
@@ -151,7 +149,6 @@ export const InternalEditor = ({
         overrides={{
           header: () => {
             const { refreshPermissions, config } = usePuck();
-            setPuckRendered(true);
```
Comment (mkilpatrick) [inline src/internal/components/InternalEditor.tsx:154]: I asked if there was a way to do this in Discord. They said we could try getting `iframeRef` from usePuck and then doing iframeRef.onload. Give that a shot to see if it allows us to remove the setTimeout

Comment (mkilpatrick) [reply src/internal/components/InternalEditor.tsx:154]: Jk, it's not available yet. Someone just put up a PR for it https://github.com/measuredco/puck/pull/652

Comment (chrisvxd) [reply src/internal/components/InternalEditor.tsx:154]: You could try `#preview-frame` to access the iframe

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**